salt/daemons/masterapi.py

```diff
@@ -836,6 +836,46 @@ def minion_runner(self, load):
         runner = salt.runner.Runner(opts)
         return runner.run()
 
+    def minion_wheel(self, load):
+        '''
+        Execute a wheel from a minion, return the wheel's function data
+        '''
+        if 'peer_wheel' not in self.opts:
+            return {}
+        if not isinstance(self.opts['peer_wheel'], dict):
+            return {}
+        if any(key not in load for key in ('fun', 'arg', 'id')):
+            return {}
+        perms = set()
+        for match in self.opts['peer_wheel']:
+            if re.match(match, load['id']):
+                # This is the list of funcs/modules!
+                if isinstance(self.opts['peer_wheel'][match], list):
+                    perms.update(self.opts['peer_wheel'][match])
+        good = False
+        for perm in perms:
+            if re.match(perm, load['fun']):
+                good = True
+        if not good:
+            # The minion is not who it says it is!
+            # We don't want to listen to it!
+            log.warning('Minion id %s is not who it says it is!', load['id'])
```

Comment:
Eh? This is a perms problem and not an identify problem, isn't it?

Reply:
yup, fixed.